### PR TITLE
docker: Use available storage space when available

### PR DESCRIPTION
**Pull request by petervo**

### CODE REVIEW

**Comment by stefwalter** [comment]
Did a related patch go into Docker?

**Comment by petervo** [comment]
Yes, https://github.com/docker/docker/commit/f4dc496d36d31cf9ca1b3508f10954066ff7f8bc

File: pkg/shell/cockpit-docker.js [hunk 1]
```diff
@@ -644,24 +644,35 @@ PageContainers.prototype = {
...
               var b_total = modules.docker.bytes_from_format(total);
 
+              // Prefer available if present as that will be accurate for
+              // sparse file based devices
+              if (avail) {
+                  b_total = modules.docker.bytes_from_format(avail);
+                  total = cockpit.format_bytes(b_used + b_total);
+                  $('#containers-storage').attr("title", "").tooltip('destroy');
+              } else {
+                  var warning = _("WARNING: Docker may be reporting the size it has alocated to it's storage pool using sparse files, not the actual space available to the underlying storage device.");
```
Comment (mvollmer) [inline pkg/shell/cockpit-docker.js]: "allocated"

File: pkg/shell/cockpit-docker.js [hunk 2]
```diff
@@ -644,24 +644,35 @@ PageContainers.prototype = {
...
+                  $('#containers-storage').attr("title", warning).tooltip();
```
Comment (mvollmer) [inline pkg/shell/cockpit-docker.js]: I get two tooltips for this.
double-tip

**Comment by mvollmer** [comment]
My two comments apply to master as well, I guess... I'll file an issue.

**Pull request closed without merging**